File: 11_search-insert-position/Binary_Search/binarysearch.cpp
#include <iostream>
int RBinarySearch(int arr[] ,int l ,int h , int target) {

    if (l > h) {
    return -1;
    }

    int mid = ( l + h ) / 2;
    if ( target == arr[mid] ) { 
        return mid;
    }
    if (target < arr[mid]) {
        h = mid - 1;
        return RBinarySearch(arr,0,h,target);
    } else {
        l = mid + 1;
        return RBinarySearch(arr,l,h,target);
    }
}


int main() {
    int arr[] = { 0, 1, 3, 4, 5,55,60,75};
    int result = RBinarySearch(arr,0,sizeof(arr)/sizeof(arr[0])-1 ,55);

    std::cout << result << "\n";
    return 0;
}

File: 11_search-insert-position/Binary_Search/binarysearch.py
def BinarySearch(nums , target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: int
    """
    l = 0
    h = len(nums) - 1

    while (l <= h):
        mid = (l + h)//2
        if (target == nums[mid]):
            return mid
        elif (target < nums[mid]):
            h  = mid - 1 
        else:
            l = mid + 1

    return -1

nums = [1,2,3,5,8,12,66,89]

result = BinarySearch(nums, target = 1)

print(result)